xdsl/backend/register_type.py

```diff
@@ -101,7 +135,9 @@ def infinite_register_prefix(cls) -> str:
     def infinite_register(cls, index: int) -> Self:
         """
         Provide the register at the given index in the "infinite" register set.
+        Index must be positive.
         """
+        assert index >= 0, f"Infinite index must be positive, got {index}."
```

Comment:
So, #3964 is obsolete now?

Reply:
Not quite, I'll reopen that one after merging this.